radixdlt-java/src/test/java/com/radixdlt/client/core/ledger/RadixParticleStoreTest.java

```diff
@@ -0,0 +1,36 @@
+package com.radixdlt.client.core.ledger;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.radixdlt.client.atommodel.accounts.RadixAddress;
+import com.radixdlt.client.core.atoms.Atom;
+import com.radixdlt.client.core.atoms.AtomObservation;
+import com.radixdlt.client.core.atoms.particles.Particle;
+import com.radixdlt.client.core.atoms.particles.SpunParticle;
+import io.reactivex.Observable;
+import io.reactivex.observers.TestObserver;
+import java.util.Collections;
+import org.junit.Test;
+
+public class RadixParticleStoreTest {
+	@Test
+	public void testParticleInstructionWithAtomDelete() {
+		RadixAddress address = mock(RadixAddress.class);
+
+		Particle particle = mock(Particle.class);
+		when(particle.getAddresses()).thenReturn(Collections.singleton(address));
+
+		Atom atom = new Atom(Collections.singletonList(SpunParticle.up(particle)));
```

Comment:
I'm a bit worried that many of our tests involve actually creating Atoms rather than mocking them. This is of course because mocking doesn't work properly on final classes, but I think we should consider some alternatives / solutions. The way it is now, if something in Atom breaks / changes, we will get many dubious test results.